test/python/result/test_result.py

```diff
@@ -133,6 +133,49 @@ def test_marginal_counts_result(self):
         self.assertEqual(marginal_counts(result, [0]).get_counts(1),
                          expected_marginal_counts_2)
 
+    def test_marginal_counts_in_place_result(self):
+        """Test that a Result object containing counts marginalizes correctly
+            when in_place = True.
+        """
```

Comment:
The first line in the test docstring should summarize the test purpose (it is printed when verbose). 
```suggestion
    def test_marginal_counts_inplace_true(self):
        """Test marginal_counts(Result, inplace = True)
        """
```

Do you mind add a test for explicit `False`?

Reply:
Sure!

Reply:
@1ucian0 please have a look at [29c857](https://github.com/Qiskit/qiskit-terra/pull/4929/commits/29c857689fb3f037c6a645423553e02d63826120)